pkg/server/smtp/handler.go

```diff
@@ -384,7 +385,16 @@ func (s *Session) readyHandler(cmd string, arg string) {
 			return
 		}
 		from := m[1]
+		s.logger.Debug().Msgf("Mail sender is %v", from)
 		localpart, domain, err := policy.ParseEmailAddress(from)
+		s.logger.Debug().Msgf("Domain sender is %v", domain)
+
+		if stringutil.SliceContains(s.Server.config.RejectOriginDomains, domain) {
```

Comment:
This should use the new method you defined above in address.go.  In the future we could add a `sender.go` or `origin.go` policy file similar to the existing `recipient.go` to encapsulate any checks we want to do, but you don't have to do this unless you want to.

Reply:
Ok  
I've choosen to create this new origin.o file.  
I could not succeed to make requested changes in address.go.